lhs.js

```diff
@@ -144,10 +141,16 @@ class DomLhs extends Lhs {
         this.selector = selector;
     }
 
+    clone(pred) {
+        let lhs = new this.constructor(this.selector);
+        lhs._predicate = pred;
```

Comment:
Since this is called simply "clone", it shouldn't also set a predicate. You could rename it "cloneWithPredicate" or something, but I recommend simply moving the predicate-setting out of it and into the caller, since a clone method is often a useful primitive and will likely prove useful in the future.